presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -856,6 +858,30 @@ public void commitInsert(ConnectorSession session, ConnectorInsertTableHandle in
             for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
                 MoreFutures.getFutureValue(fileRenameFuture, PrestoException.class);
             }
+
+            if (handle.getWritePath().isPresent()) {
```

Comment:
Please check that `getWritePath()` does not equal `getTargetPath()`, which can happen on S3 when creating a new table. We use targetPath as writePath when creating a table on S3, but empty when modifying an existing table in S3. Javadoc for LocationService.writePath briefly explains why we did this.